### ISSUE TITLE
App duplicates

### ISSUE THREAD

**Issue by voroyam** [owncloud/core#39930]
### Steps to reproduce
1. install oC 10.9.1 complete
2. Update or install an app via the market app in the browser
3. execute `occ a:l -m`

### Expected behaviour
apps are listed once

### Actual behaviour
duplicates:

```
Enabled:
  - activity 2.7.0
  - activity 2.7.0
  - comments 0.3.0
  - configreport 0.2.0
  - dav 0.7.0
  - federatedfilesharing 0.5.0
  - federation 0.1.0
  - files 1.5.2
  - files_clipboard 1.0.3
  - files_external 0.8.0
  - files_mediaviewer 1.0.5
  - files_pdfviewer 1.0.1
  - files_sharing 0.14.0
  - files_texteditor 2.4.1
  - files_trashbin 0.9.1
  - files_versions 1.3.0
  - firstrunwizard 1.2.0
  - guests 0.9.3
  - guests 0.9.3
  - market 0.6.3
  - market 0.6.3
  - notifications 0.5.4
  - oauth2 0.5.2
  - oauth2 0.5.2
  ```

I looked in the folders: apps and apps-external both have the newest version of the app, so here are my questions:

1. why can you install an app that is already there because you have the complete bundle?
2. why is it listed twice in the `occ app:list` -m but only once in the `occ app:list` commands?
3. Notable is that apps which are not part of the bundle, like files_clipboard, are listed once, so the problem is when an app is there in the apps folder, and you install it via market app, but then - why is it not listed as an installed app?

### Server configuration
**Operating system**:
Ubuntu 20.04

**Web server:**
Apache2

**Database:**
Mariah DB

**PHP version:**
7.4

**ownCloud version:** (see ownCloud admin page)
10.9.1

**Updated from an older ownCloud or fresh install:**
fresh

**Where did you install ownCloud from:**
owncloud.com

**The content of config/config.php:**

```
'apps_paths' =>
  array (
    0 =>
    array (
      'path' => '/var/www/owncloud/apps',
      'url' => '/apps',
      'writable' => false,
    ),
    1 =>
    array (
      'path' => '/var/www/owncloud/apps-external',
      'url' => '/apps-external',
      'writable' => true,
    ),
  ),
```

**List of activated apps:**

```
root@session:/var/www/owncloud# occ a:l
Enabled:
  - activity:
    - Version: 2.7.0
    - Path: /var/www/owncloud/apps-external/activity
  - comments:
    - Version: 0.3.0
    - Path: /var/www/owncloud/apps/comments
  - configreport:
    - Version: 0.2.0
    - Path: /var/www/owncloud/apps/configreport
  - dav:
    - Version: 0.7.0
    - Path: /var/www/owncloud/apps/dav
  - federatedfilesharing:
    - Version: 0.5.0
    - Path: /var/www/owncloud/apps/federatedfilesharing
  - federation:
    - Version: 0.1.0
    - Path: /var/www/owncloud/apps/federation
  - files:
    - Version: 1.5.2
    - Path: /var/www/owncloud/apps/files
  - files_clipboard:
    - Version: 1.0.3
    - Path: /var/www/owncloud/apps-external/files_clipboard
  - files_external:
    - Version: 0.8.0
    - Path: /var/www/owncloud/apps/files_external
  - files_mediaviewer:
    - Version: 1.0.5
    - Path: /var/www/owncloud/apps/files_mediaviewer
  - files_pdfviewer:
    - Version: 1.0.1
    - Path: /var/www/owncloud/apps/files_pdfviewer
  - files_sharing:
    - Version: 0.14.0
    - Path: /var/www/owncloud/apps/files_sharing
  - files_texteditor:
    - Version: 2.4.1
    - Path: /var/www/owncloud/apps/files_texteditor
  - files_trashbin:
    - Version: 0.9.1
    - Path: /var/www/owncloud/apps/files_trashbin
  - files_versions:
    - Version: 1.3.0
    - Path: /var/www/owncloud/apps/files_versions
  - firstrunwizard:
    - Version: 1.2.0
    - Path: /var/www/owncloud/apps/firstrunwizard
  - guests:
    - Version: 0.9.3
    - Path: /var/www/owncloud/apps/guests
  - market:
    - Version: 0.6.3
    - Path: /var/www/owncloud/apps-external/market
  - notifications:
    - Version: 0.5.4
    - Path: /var/www/owncloud/apps/notifications
  - oauth2:
    - Version: 0.5.2
    - Path: /var/www/owncloud/apps/oauth2
  - provisioning_api:
    - Version: 0.5.0
    - Path: /var/www/owncloud/apps/provisioning_api
  - systemtags:
    - Version: 0.3.0
    - Path: /var/www/owncloud/apps/systemtags
  - templateeditor:
    - Version: 0.4.0
    - Path: /var/www/owncloud/apps/templateeditor
  - twofactor_totp:
    - Version: 0.7.4
    - Path: /var/www/owncloud/apps-external/twofactor_totp
  - updatenotification:
    - Version: 0.2.1
    - Path: /var/www/owncloud/apps/updatenotification
Disabled:
  - admin_audit:
    - Path: /var/www/owncloud/apps/admin_audit
  - announcementcenter:
    - Path: /var/www/owncloud/apps/announcementcenter
  - customgroups:
    - Path: /var/www/owncloud/apps/customgroups
  - encryption:
    - Path: /var/www/owncloud/apps/encryption
  - enterprise_key:
    - Path: /var/www/owncloud/apps/enterprise_key
  - external:
    - Path: /var/www/owncloud/apps/external
  - files_antivirus:
    - Path: /var/www/owncloud/apps/files_antivirus
  - files_classifier:
    - Path: /var/www/owncloud/apps/files_classifier
  - files_external_dropbox:
    - Path: /var/www/owncloud/apps/files_external_dropbox
  - files_external_ftp:
    - Path: /var/www/owncloud/apps/files_external_ftp
  - files_ldap_home:
    - Path: /var/www/owncloud/apps/files_ldap_home
  - files_lifecycle:
    - Path: /var/www/owncloud/apps/files_lifecycle
  - firewall:
    - Path: /var/www/owncloud/apps/firewall
  - graphapi:
    - Path: /var/www/owncloud/apps/graphapi
  - metrics:
    - Path: /var/www/owncloud/apps/metrics
  - openidconnect:
    - Path: /var/www/owncloud/apps/openidconnect
  - password_policy:
    - Path: /var/www/owncloud/apps/password_policy
  - ransomware_protection:
    - Path: /var/www/owncloud/apps/ransomware_protection
  - sharepoint:
    - Path: /var/www/owncloud/apps/sharepoint
  - systemtags_management:
    - Path: /var/www/owncloud/apps/systemtags_management
  - theme-enterprise:
    - Path: /var/www/owncloud/apps/theme-enterprise
  - user_external:
    - Path: /var/www/owncloud/apps/user_external
  - user_ldap:
    - Path: /var/www/owncloud/apps/user_ldap
  - user_shibboleth:
    - Path: /var/www/owncloud/apps/user_shibboleth
  - web:
    - Path: /var/www/owncloud/apps/web
  - windows_network_drive:
    - Path: /var/www/owncloud/apps/windows_network_drive
  - wopi:
    - Path: /var/www/owncloud/apps/wopi
  - workflow:
    - Path: /var/www/owncloud/apps/workflow
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Chrome

**Operating system:**
MacOS

#### ownCloud log (data/owncloud.log)
```
nothing in the logs
```

**Comment by voroyam** [comment]
@IljaN

**Comment by jnweiger** [comment]
Still seen with 10.10.0 RC2

occ a:l -m
...
  - web 5.5.0
  - web 5.5.0
  - windows_network_drive 2.1.1
  - windows_network_drive 2.1.1

**Comment by phil-davis** [comment]
The "accident" of listing twice with the `--minimal` option is easily fixed - see PR #40081 

A side-effect of the bug was that the user could see that an app existed twice. If the app had different versions in each place, then the listing would show 2 different versions numbers. That might be useful information for an admin who is trying to track down a problem.

But the accidental duplicate listing did not specify which entry came from `apps` and which from `apps-external`. So an admin would not have quite had all the information that they might like.

Maybe there should be some extra things available:
1) `apps:list` should always show the version that will be used by the running system, with the relevant version number and path
2) `apps:list` should have an options to display all "copies" of apps, with the individual version and path of each "copy" of an app. That would help admins see what they have got on their system.

**Comment by jnweiger** [comment]
Confirmed fixed in 10.11.0-alpha.2
